R/utils.R

```diff
@@ -1,40 +1,274 @@
-#' Generate simulation times and the indices
-#' of time points to return in sumulation functions.
-
-#' @param t.start Numeric scalar. Initial time of the simulation, default 0.
-#' @param t.end Numeric scalar. Final time of the simulation, default 1000.
-#' @param t.step Numeric scalar. Interval between simulation steps, default 0.1.
-#' @param t.keep Integer scalar. Number of evenly distributed time points to keep, default 100.
+#' @title Generate simulation times and the indices of time points to return
+#' in simulation functions.
+#'
+#' @param t_start Numeric scalar indicating the initial time of the simulation.
+#' (default: \code{t_start = 0})
+#' @param t_end Numeric scalar indicating the final time of the dimulation
+#' (default: \code{t_end = 1000})
+#' @param t_step Numeric scalar indicating the interval between simulation steps
+#' (default: \code{t_step = 0.1})
+#' @param t_store Integer scalar indicating the number of evenly distributed
+#' time points to keep (default: \code{t_store = 100})
 #'
-#' @return list containing simulation times (t.sys) and the indices to keep.\
+#' @return lists containing simulation times (t_sys) and the indices to keep.
+#' @examples
+#' Time <- simulationTimes(t_start = 0, t_end = 100, t_step = 0.5,
+#'     t_store = 100)
+#' DefaultTime <- simulationTimes(t_end = 1000)
+#'
+#' @importFrom utils combn
+#' @importFrom stats cov
+#' @importFrom stats rbinom
+#' @importFrom reshape2 melt
+#' @import ggplot2
+#' 
 #' @keywords internal
 #' @export
 simulationTimes <- function(t_start = 0, t_end = 1000,
             t_step = 0.1, t_store = 1000){
         t_total <- t_end-t_start
         t_sys <- seq(t_start, t_end, by = t_step)
-        t_index <- seq(1, length(t_sys)-1, by=round(length(t_sys)/t_store))
-        return(list("t_sys" = t_sys, "t_index" = t_index))
+    t_index <- seq(1, length(t_sys)-1, by=floor(length(t_sys)/t_store))
+    return(list("t_sys" = t_sys, "t_index" = t_index[seq_len(t_store)]))
 }
 
-
+#' @title check whether a number is a positive integer
+#' @param x Numeric number to test
+#' @param tol Numeric tolerance of detection
+#' @return A logical value: whether the number is a positive integer.
+#' @export
 isPositiveInteger <- function(x, tol = .Machine$double.eps^0.5) {
         return(abs(x - round(x)) < tol && x > 0)
 }
 
+#' @title check whether a number is zero or positive integer
+#' @param x Numeric number to test
+#' @param tol Numeric tolerance of detection
+#' @return A logical value: whether the number is zero or positive integer.
+#' @export
 isZeroOrPositiveInteger <- function(x, tol = .Machine$double.eps^0.5) {
     return(abs(x - round(x)) < tol && x >= 0)
 }
```

Comment:
Does this function has to be exported? Typically we try to avoid exporting auxiliary functions that are internally used by functions. This has many benefits. I suggest to remove "@export" and also check all other functions for the same.

Another convention is to use dot name, like `.isNonnegativeInteger`